config/sevntu-suppressions.xml

```diff
@@ -39,6 +39,7 @@
 
   <!-- Suppressed alternative method of pulling partial configuration from file -->
   <suppress checks="CheckstyleTestMakeup" files="[\\/]RightCurlyTest\.java"/>
+  <suppress checks="CheckstyleTestMakeup" files="[\\/]ClassAnnotationsTest\.java"/>
```

Comment:
It was checkstyle violation.
`ERROR] [checkstyle] [ERROR] C:\projects\checkstyle\src\it\java\com\google\checkstyle\test\chapter4formatting\rule4852classannotations\ClassAnnotationsTest.java:51:46: DefaultConfiguration was not assigned by an allowed create configuration method. [CheckstyleTestMakeup]`

When I removed such suppression, `mvn verify` passed, so it means this suppression is not required.

Reply:
Let's try to remove it to let me see this anomaly

Reply:
done

Reply:
Failing CI: https://ci.appveyor.com/project/Checkstyle/checkstyle/builds/49952243/job/96b020g8euk1q1ld#L311

Reply:
It is weird why only windows is affected, Linux execution is green.
We will deal with this in separate issue.

Reply:
Please restore suppression.

Reply:
done

Reply:
Violation looks correct. `createTreeWalkerConfig` is not an allowed method. You can see it is extremely rare for us to use it in our code base.

I mentioned before this method of picking and pulling configs just doesn't seem like the best approach. How will we validate the checks we manually pick are the same ones we have identified in the sections and that we didn't miss one or add one? The whole reason we had a 1 to 1 check/test is this was a way to validate we were testing against what we are saying.

Is there a reason we just don't apply the whole google config against the input file?

Reply:
https://github.com/checkstyle/checkstyle/issues/14939

Reply:
My comment is not on the inconsistent CI, but on the code itself.

Reply:
>Is there a reason we just don't apply the whole google config against the input file?

It might be too much violations, or too hard to make Input.
I remember we tried at initial year to make it and there was some mess. 
I am not sure on this.
It is easy to make 0 violations Inputs, and with violations might have bunch unrelated violations.

Reply:
> How will we validate the checks we manually pick are the same ones we have identified in the sections and that we didn't miss one or add one? The whole reason we had a 1 to 1 check/test is this was a way to validate we were testing against what we are saying in the documentation.


You have a valid point. We cannot test if all the mentioned modules in a section are used in the test or if any module is missing in the test. This approach focuses on combining different modules into one test so we can test the whole section in one go. For this, I guess we need to manually ensure that all the modules are valid and no modules is left behind.

Reply:
Manual is ok for a time, but I see this as being a concern. All reviewers must ensure every tests matches google doc each time an update is done.

I don't see how it is bad we have unrelated violations if we apply the whole google config against input files. The whole point of this is we are showing we follow google's guide, so any and all violations which are displayed should be part of what we are showcasing. If a violation is "wrong", then this would be a bug in our config and would need to be fixed.

Reply:
>  My motto is if you don't enforce something by CI, someone will find a way to do something the wrong way. 

Why CI? We can write test in XdocPagesTest so why we should enforce something by CI?

> I don't see how it is bad we have unrelated violations if we apply the whole google config against input files. The whole point of this is we are showing we follow google's guide, so any and all violations which are displayed should be part of what we are showcasing. If a violation is "wrong", then this would be a bug in our config and would need to be fixed.

You're correct but checking the input against whole config will require us to take small things into account like adding javadocs, maintaining indentation ( google uses 2 spaces for indentation ), etc. This can become annoying to do as we need to ensure all these small details in every tests we write. Romani also said in the discord that, earlier we tried this approach but it was a mess or something like that. IMO we should properly discuss this before moving forward. I'm fine with abandoning the idea of validating the modules being used in the test. We can implement it after we come to a solid decision.

Reply:
> We can write test in XdocPagesTest 

Tests are run by CI, so if it is in a test, it is enforced. I wasn't implying writing in the CI's YML directly, sorry for that confusion.

> checking the input against whole config will require us to take small things into account like adding javadocs, maintaining indentation ( google uses 2 spaces for indentation ), etc. This can become annoying to do as we need to ensure all these small details in every tests we write. 

Yes, this is why I was suggesting this not be part of this GSoC. It does also seem odd to show these files off as following GSoC, when as you bring up, they do not follow them. :)

> I'm fine with abandoning the idea of validating the modules being used in the test. We can implement it after we come to a solid decision.

I agree.